impl.go

```diff
@@ -19,29 +20,36 @@ import (
 	"golang.org/x/tools/imports"
 )
 
-const usage = `impl <recv> <iface>
+const usage = `impl [-dir directory] <recv> <iface>
 
 impl generates method stubs for recv to implement iface.
 
 Examples:
 
 impl 'f *File' io.Reader
 impl Murmur hash.Hash
+impl Murmur hash.Hash -dir $GOPATH/src/github.com/josharian/impl
 
 Don't forget the single quotes around the receiver type
 to prevent shell globbing.
 `
 
+var (
+	sourceDirectory = flag.String("dir", "", "package source directory")
```

Comment:
Nit: please name this `flagSourceDirectory` or `flagSrcDir`. It helps highlight that it is a flag value, even far from its declaration.

If you can, it'd be nice to add a short clause hinting to the reader why/when they would want this, maybe something like `"package source directory, useful for vendored code"`.